front/src/common/authorization/components/GrantsManagerSubjects.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState } from 'react';
+
+import { ComboBox, Select } from '@osrd-project/ui-core';
+import cx from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import type { Grant, ResourceType, SubjectItemWithGrant } from 'common/api/mock/mockEditoastApi';
+import useSearchUser from 'common/useSearchUser';
+import { DEFAULT_GRANT } from 'modules/infra/consts';
+import generateGrantSelectProps from 'modules/infra/utils/generateGrantSelectProps';
+
+import useSubjectsResourceGrants from '../hooks/useSubjectsResourceGrants';
+
+type GrantsManagerSubjectsProps = {
+  resourceId: number;
+  resourceType: ResourceType;
+  userGrant: string;
```

Comment:
We get a type `Grant` from the endpoint. Might as well use it.
```suggestion
  userGrant: Grant;
```

Reply:
We can also use `Grant` in `checkPrivileges` for param `userGrant`